src/sentry_core.c

```diff
@@ -1169,3 +1169,69 @@ sentry_clear_crashed_last_run(void)
     sentry__options_unlock();
     return success ? 0 : 1;
 }
+
+void
+sentry_capture_minidump(
+    const char *dump_path, sentry_value_t event, int removeDumpOnSend)
+{
+    sentry_path_t *sentry_dump_path
+        = sentry__path_from_str_n(dump_path, strlen(dump_path));
+
+    if (sentry_dump_path == NULL) {
+        SENTRY_WARN(
+            "'sentry_capture_minidump' Failed due to null path to minidump");
+        return;
+    }
+
+    SENTRY_WITH_OPTIONS (options) {
+        sentry_envelope_t *envelope
+            = NULL;
+        sentry__ensure_event_id(event, NULL);
+        envelope = sentry__envelope_new();
```

Comment:
I was probably not clear about this, but I meant not to split but to have declaration and initialization in one:
```suggestion
        sentry__ensure_event_id(event, NULL);
        sentry_envelope_t *envelope = sentry__envelope_new();
```